pkg/controllers/override/imagepatch.go

```diff
@@ -0,0 +1,203 @@
+		}
+		index++
+	}
+
+	return patches, nil
+}
+
+// parsePatchesFromImagePath applies overrider to image value on the specified path,
+// and generates a patch which is stored in patchMap
+func parsePatchesFromImagePath(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	imageOverrider *fedcorev1a1.ImageOverrider,
+	patchMap map[string]string,
+) error {
+	imagePath := imageOverrider.ImagePath
+	imageValue := patchMap[imagePath]
+
+	// get image value
+	if imageValue == "" {
+		// get source obj
+		sourceObj, err := fedObject.GetSpec().GetTemplateAsUnstructured()
+		if err != nil {
+			return fmt.Errorf("failed to get sourceObj from fedObj: %w", err)
+		}
+
+		imageValue, err = ParseValueFromUnstructuredObj(sourceObj, imageOverrider.ImagePath)
+		if err != nil {
+			return fmt.Errorf("failed to parse image value from unstructured obj: %w", err)
+		}
+	}
+
+	// apply image override
+	newImageValue, err := applyImageOverride(imageOverrider.ImagePath, imageValue, imageOverrider)
+	if err != nil {
+		return fmt.Errorf("failed apply image override: %w", err)
+	}
+	patchMap[imagePath] = newImageValue
+
+	return nil
+}
+
+// parsePatchesFromWorkload applies overrider to image value on the default path of workload,
+// and generates a patch which is stored in patchMap
+func parsePatchesFromWorkload(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	imageOverrider *fedcorev1a1.ImageOverrider,
+	patchMap map[string]string,
+) error {
+	// get pod spec from fedObj
+	sourceObj, err := fedObject.GetSpec().GetTemplateAsUnstructured()
+	if err != nil {
+		return fmt.Errorf("failed to get sourceObj from fedObj: %w", err)
+	}
+	gvk := sourceObj.GetObjectKind().GroupVersionKind()
+	podSpec, err := podutil.GetResourcePodSpec(fedObject, gvk)
+	if err != nil {
+		return fmt.Errorf("failed to get podSpec from sourceObj: %w", err)
+	}
+
+	// convert user-specified containerNames into set
+	specifiedContainers := sets.New[string]()
+	for _, name := range imageOverrider.ContainerNames {
+		if name != "" {
+			specifiedContainers.Insert(name)
+		}
+	}
+
+	for index, container := range podSpec.Containers {
+		// if specified container list is empty, the image override rule applies to all containers.
+		if len(imageOverrider.ContainerNames) == 0 || specifiedContainers.Has(container.Name) {
+			imagePath, err := generateImagePath(gvk, index)
+			if err != nil {
+				return err
+			}
+			imageValue := patchMap[imagePath]
+			if imageValue == "" {
+				imageValue = container.Image
+			}
+
+			newImageValue, err := applyImageOverride(imagePath, imageValue, imageOverrider)
+			if err != nil {
+				return fmt.Errorf("failed to apply image override: %w", err)
+			}
+			patchMap[imagePath] = newImageValue
+		}
+	}
+
+	return nil
+}
+
+// applyImageOverride applies override to oldImage to generate a new value
+func applyImageOverride(imagePath, oldImage string, imageOverrider *fedcorev1a1.ImageOverrider) (string, error) {
+	if !strings.HasPrefix(imagePath, pathSeparator) {
```

Comment:
This check seems a bit out-of-place. Any reason it's performed here instead of earlier?

Reply:
yes, currently we only need to verify the image path specified by user. The code here has been placed in the `parsePatchesFromImagePath` function.